### ISSUE TITLE
Tokio dependencies

### ISSUE THREAD

**Issue by gmosx** [http-rs/tide#427]
I tried to build the simple 'hello-world' example from the README. I see that multiple `tokio` dependencies are added, e.g.

```
...
 Compiling tokio-executor v0.1.10
 Compiling tokio-current-thread v0.1.7
 Compiling tokio-timer v0.2.13
 Compiling tokio-io v0.1.13
 Compiling tokio-threadpool v0.1.18
 Compiling tokio-reactor v0.1.12
 Compiling tokio-tcp v0.1.4
 Compiling tokio v0.1.22
...
```

Is this intentional?

**Comment by minioin** [comment]
I think it's consequential, different dependencies pulling in different versions.

**Comment by yoshuawuyts** [comment]
This should be resolved by #414; we haven't fully optimized our dependencies yet, and there are still some odd edge cases like these present.

**Comment by pickfire** [comment]
Does not this increase the build time?
